app/uk/gov/hmrc/gform/testonly/filter/UserInputs.scala

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.testonly.filter
+
+import cats.implicits._
+import uk.gov.hmrc.govukfrontend.views.Aliases.{ ErrorMessage, Fieldset, HtmlContent, InputItem, Legend }
+import uk.gov.hmrc.govukfrontend.views.viewmodels.dateinput.DateInput
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content
+import play.api.mvc.QueryStringBindable
+// import uk.gov.hmrc.govukfrontend.views.html.components._
+import uk.gov.hmrc.govukfrontend.views.html.components._
+import play.twirl.api.Html
+
+case class DateTimeUserInput(
+  day: Option[String],
+  month: Option[String],
+  year: Option[String],
+  hour: Option[String],
+  minute: Option[String]
+)
+
+object DateTimeUserInput {
+
+  private val govukErrorMessage: GovukErrorMessage = new GovukErrorMessage()
+  private val govukHint: GovukHint = new GovukHint()
+  private val govukLabel: GovukLabel = new GovukLabel()
+  private val govukFieldset: GovukFieldset = new GovukFieldset()
+  private val govukInput: GovukInput = new GovukInput(govukErrorMessage, govukHint, govukLabel)
+  private val govukDateInput: GovukDateInput = new GovukDateInput(
+    govukErrorMessage,
+    govukHint,
+    govukFieldset,
+    govukInput
+  )
```

Comment:
This should really be done in twirl templates if possible. Is there particular reason why is it here?